src/spatial/collision.js

```diff
@@ -124,14 +119,13 @@ Crafty.c("Collision", {
         }
 
         // Finally, assign the hitbox, and attach it to the "Collision" entity
-        this.map = poly;
+        this.map = polygon;
         this.attach(this.map);
         this.map.shift(this._x, this._y);
```

Comment:
Is it okay what shifting is made after attaching and `NewHitbox` event triggering?